Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm

```diff
@@ -15048,9 +15093,23 @@ @implementation WKFoundTextRange
 
 + (WKFoundTextRange *)foundTextRangeWithWebFoundTextRange:(WebKit::WebFoundTextRange)webRange
 {
-    auto range = adoptNS([[WKFoundTextRange alloc] init]);
-    [range setLocation:webRange.location];
-    [range setLength:webRange.length];
+    RetainPtr range = WTF::switchOn(webRange.data,
```

Comment:
This looks like one of those thingswhere you could expect to get away without a return type hint, but maybe it was necessary?

Reply:
Yeah, the compiler wasn't smart enough to figure out that the two classes were derived from the same class.